src/firestore-adapter.ts

```diff
@@ -653,50 +666,48 @@ export class FirestoreAdapter implements IDatabaseAdapter {
         callback(error, cursor);
       }
     });
-
-    this._userCursor = cursor;
-  }
-
-  /**
-   * Callback listener for `child_added` and `child_changed` events on `users` node of Firebase ref.
-   * @param childSnap - JSON serializable data snapshot of the child.
-   */
-  protected _childChanged(childSnap: firebase.database.DataSnapshot): void {
-    if (this._zombie) {
-      // just in case we were cleaned up before we got the users data.
-      return;
-    }
-
-    const userId = childSnap.key as string;
-    const userData = childSnap.val() as FirebaseCursorDataType;
-
-    this._trigger(
-      FirebaseAdapterEvent.CursorChange,
-      userId,
-      userData.cursor,
-      userData.color,
-      userData.name
+    this._firestoreUserRef.set(
+      { lastUpdated: Date.now(), cursor: { ...cursorData } },
+      { merge: true }
     );
-  }
 
-  /**
-   * Callback listener for `child_removed` events on `users` node of Firebase ref.
-   * @param childSnap - JSON serializable data snapshot of the child.
-   */
-  protected _childRemoved(childSnap: firebase.database.DataSnapshot): void {
-    const userId = childSnap.key as string;
-    this._trigger(FirebaseAdapterEvent.CursorChange, userId, null);
+    this._userCursor = cursor;
   }
 
   /**
    * Attach listeners for `child_added`, `child_changed` and `child_removed` event on `users` node of Firebase ref.
    */
   protected _monitorCursors(): void {
-    const usersRef = this._databaseRef!.child("users");
-
-    this._firebaseOn(usersRef, "child_added", this._childChanged, this);
-    this._firebaseOn(usersRef, "child_changed", this._childChanged, this);
-    this._firebaseOn(usersRef, "child_removed", this._childRemoved, this);
+    const firestoreUsersRef = this._firestoreRef!.collection("users");
+    firestoreUsersRef.onSnapshot((snapshot) => {
+      snapshot.docChanges().forEach((change) => {
+        if (change.type === "added" || change.type === "modified") {
+          const userId = change.doc.id;
+          const userData = change.doc.data() as FirebaseCursorDataType;
+          if (
+            userData["lastUpdated"] &&
+            Date.now() - userData["lastUpdated"] <
```

Comment:
This will not remove expired cursors since those are not `modified` anymore. Rather we should watch a filter query and get the cursor removed from results. Query can be as follows:

```javascript
const firestoreUsersRef = this._firestoreRef!.collection("users").where(
    "lastUpdated",
    ">=",
    Date.now() - FirestoreAdapter.MAX_CURSOR_SYNC_TIME
);
```